makiflow/layers/initializers.py

```diff
@@ -0,0 +1,443 @@
+# Copyright (C) 2020  Igor Kilbas, Danil Gribanov, Artem Mukhin
+#
+# This file is part of MakiFlow.
+#
+# MakiFlow is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# MakiFlow is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Foobar.  If not, see <https://www.gnu.org/licenses/>.
+
+import numpy as np
+from abc import abstractmethod, ABC
+
+# Some initialize methods
+# Initializations define the way to set the initial random weights of MakiFlow layers.
+
+# (in_f, out_f), in most cases
+NUM_OF_PARAMS_DENSE = 2
```

Comment:
Better to get rid of dependence on exact weights' shape.
General shape is `[nd0, nd1, ..., ndn]` where `ndn` corresponds to the output features dimensions and the rest are input features dimensions.
In most of the cases weights are being normalized by the square root of product of input dimensions:
`scale = 1 / sqrt(nd0*nd1*...*nd(n-1))`
`scale = 1 / np.sqrt(np.prod(shape[:-1]))`.
Try to formulate weights generation in a more general manner.